ocs_ci/helpers/odf_cli.py

```diff
@@ -0,0 +1,119 @@
+import os
+
+from stat import S_IEXEC
+from logging import getLogger
+from typing import Union
+
+from ocs_ci.ocs.exceptions import NotSupportedException
+from ocs_ci.utility.version import get_semantic_ocs_version_from_config, VERSION_4_15
+from ocs_ci.utility.utils import exec_cmd
+from ocs_ci.framework import config
+from ocs_ci.deployment.ocp import download_pull_secret
+
+
+log = getLogger(__name__)
+
+
+class ODFCLIRetriever:
+    def __init__(self):
+        self.semantic_version = get_semantic_ocs_version_from_config()
+        self.local_cli_path = os.path.join(config.RUN["bin_dir"], "odf-cli")
+
+    def retrieve_odf_cli_binary(self):
+        """
+        Download and set up the ODF-CLI binary.
+
+        Raises:
+            NotSupportedException: If ODF CLI is not supported on the current version or deployment.
+            AssertionError: If the CLI binary is not found or not executable.
+        """
+        self._validate_odf_cli_support()
+
+        image = self._get_odf_cli_image()
+        self._extract_cli_binary(image)
+        self._set_executable_permissions()
+        self._verify_cli_binary()
+        self.add_cli_to_path()
+
+    def _validate_odf_cli_support(self):
+        if self.semantic_version < VERSION_4_15:
+            raise NotSupportedException(
+                f"ODF CLI tool not supported on ODF {self.semantic_version}"
+            )
+
+    def _get_odf_cli_image(self, build_no: str = None):
+        if build_no:
+            return f"quay.io/rhceph-dev/odf4-odf-cli-rhel9:{build_no}"
```

Comment:
please save the url as a constant so if changed we will be able to adjust it in a single place

Reply:
fixed